components/SnippetForm.js

```diff
@@ -0,0 +1,108 @@
+import { useState } from "react";
+import styled from "styled-components";
+import useSWR from "swr";
+
+function SnippetForm() {
+  const { mutate } = useSWR(`api/snippets`);
+
+  const [submitted, setSubmitted] = useState(false);
+  const [error, setError] = useState(null);
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const snippetData = Object.fromEntries(formData);
+
+    const response = await fetch("/api/snippets", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(snippetData),
+    });
+    event.target.reset();
+    event.target.elements.name.focus();
+    if (response.ok) {
+      setSubmitted(true);
+      setError(null);
+      mutate();
+    } else {
+      const data = await response.json();
+      setError(data.error);
+    }
+  }
+
+  return (
+    <>
+      <StyledForm onSubmit={handleSubmit}>
+        <h2>Add new Snippet</h2>
+        <label for="name">Name:</label>
```

Comment:
this need to be htmlFor

```suggestion
        <label htmlFor="name">Name:</label>
```